src/main/java/org/ehrbase/fhirbridge/ehr/converter/TherapyCompositionConverter.java

```diff
@@ -0,0 +1,264 @@
+        result.setComposer(new PartySelf());
+
+        return result;
+    }
+
+    private NameDerProzedurDefiningCode mapNameDerProzedur(Procedure procedure) throws UnprocessableEntityException {
+        Coding coding = procedure.getCode().getCoding().get(0);
+
+        if (coding.getSystem().equals(SNOMED_SYSTEM) && nameDerProzedurMap.containsKey(coding.getCode())) {
+            return nameDerProzedurMap.get(coding.getCode());
+        } else {
+            throw new UnprocessableEntityException("Invalid name of procedure");
+        }
+    }
+
+    private void mapDone(Procedure procedure, GECCOProzedurComposition composition) {
+
+        ProzedurAction durchgefuehrteProzedur = new ProzedurAction();
+
+        try {
+
+            durchgefuehrteProzedur.setNameDerProzedurDefiningCode(mapNameDerProzedur(procedure));
+
+            if (durchgefuehrteProzedur.getNameDerProzedurDefiningCode().equals(NameDerProzedurDefiningCode.PLAIN_RADIOGRAPHY)) {
+                // Map body site for PLAIN_RADIOGRAPHY
+
+                Coding bodySiteCoding = procedure.getBodySite().get(0).getCoding().get(0);
+
+                if (bodySiteCoding.getSystem().equals(SNOMED_SYSTEM) &&
+                        koerperstelleMap.containsKey(bodySiteCoding.getCode())) {
+                    durchgefuehrteProzedur.setKoerperstelleDefiningCode(koerperstelleMap.get(bodySiteCoding.getCode()));
+                } else {
+                    throw new UnprocessableEntityException("Invalid body site for PLAIN_RADIOGRAPHY");
+                }
+            } else if (durchgefuehrteProzedur.getNameDerProzedurDefiningCode().equals(NameDerProzedurDefiningCode.ARTIFICIAL_RESPIRATION_PROCEDURE)) {
+                // Map Medizingeraet for RESP
+                Coding usedCodeCoding = procedure.getUsedCode().get(0).getCoding().get(0);
+
+                if (usedCodeCoding.getSystem().equals(SNOMED_SYSTEM) && geraetenameMap.containsKey(usedCodeCoding.getCode())) {
+
+                    MedizingeraetCluster medizingeraetCluster = new MedizingeraetCluster();
+
+                    medizingeraetCluster.setGeraetenameDefiningCode(geraetenameMap.get(usedCodeCoding.getCode()));
+
+                    durchgefuehrteProzedur.setMedizingeraet(new ArrayList<>());
+                    durchgefuehrteProzedur.getMedizingeraet().add(medizingeraetCluster);
+                } else {
+                    throw new UnprocessableEntityException("Invalid medical device code");
+                }
+            }
+
+            durchgefuehrteProzedur.setArtDerProzedurDefiningCode(composition.getKategorie().get(0).getValue());
+
+
+            if (!procedure.getExtension().isEmpty()) {
+                for (Extension extension : procedure.getExtension()) {
+                    if (extension.getValue() instanceof Coding) {
+                        durchgefuehrteProzedur.setDurchfuehrungsabsichtValue(((Coding) extension.getValue()).getDisplay());
+                        break;
```

Comment:
Try to avoid break, i would suggest making a method using a return extension.getValue().getDisplay().

Reply:
I created a new method